src/path-transformers.ts

```diff
@@ -44,34 +44,41 @@
 
 import * as path from 'path';
 
-export function urlFromPath(root: string, filepath: string) {
-  if (!filepath.startsWith(root)) {
-    throw new Error(`file path is not in root: ${filepath} (${root})`);
-  }
+export function isPlatformWindows(): boolean {
+  return /^win/.test(process.platform);
+}
 
-  // On windows systems, convert filesystem path to URL by replacing slashes
-  const isPlatformWin = /^win/.test(process.platform);
-  const isExtendedLengthPath = /^\\\\\?\\/.test(filepath);
-  const hasNonAscii = /[^\x00-\x80]+/.test(filepath);
-  if (isPlatformWin && !isExtendedLengthPath && !hasNonAscii) {
-    return path.win32.relative(root, filepath).replace(/\\/g, '/');
-  }
+export function pathFromUrl(root: string, url: string) {
+  return platformifyPath(decodeURI(
+      path.posix.join(posixifyPath(root), path.posix.join('/', url))));
+}
 
-  // Otherwise, just return the relative path between the two
-  return path.relative(root, filepath);
+export function platformifyPath(filepath: string): string {
+  // Replaces all / with \ on win32.  Otherwise this is a noop.
+  // TODO(usergenic): Should we produce an "extended-length path" in win32 if
+  // the path length is over 259 on win32?
+  return filepath.replace(/\//g, path.sep);
```

Comment:
Why not `path.normalize(filepath)`?

```javascript
> require('path').win32.normalize('./foo/bar/baz')
'foo\\bar\\baz'
```

It doesn't seem like it damages paths in any way that we'd be harmed by for any of the cases I tested:

```
   ./foo         → foo
   ../foo        → ../foo
   ../foo/       → ../foo/
   ../foo/../bar → ../bar
```